src/gsnClient/gsnClient.ts

```diff
@@ -28,22 +43,40 @@ const updateConfig = async (
   config: NetworkConfig,
   transaction: GsnTransactionDetails
 ) => {
-  const { data } = await axios.get(`${config.gsn.relayUrl}/getaddr`, {
+  const response = await axios.get(`${config.gsn.relayUrl}/getaddr`, {
     headers: authHeader(config),
   });
-  //get current relay worker address from relay server config
-  config.gsn.relayWorkerAddress = data.relayWorkerAddress;
-
-  //get accepted fees from server
-  transaction.maxPriorityFeePerGas = data.minMaxPriorityFeePerGas;
-  // if chainId is 285252, use minMaxPriorityFeePerGas, otherwise use minMaxFeePerGas as minMaxFeePerGas on mumbai server set to 16 tx will fail
-  transaction.maxFeePerGas =
-    config.gsn.chainId === '80001'
-      ? data.minMaxPriorityFeePerGas
-      : data.maxMaxFeePerGas.toString();
+  const serverConfigUpdate = response.data as GsnServerConfigPayload;
+
+  config.gsn.relayWorkerAddress = serverConfigUpdate.relayWorkerAddress;
+
+  setGasFeesForTransaction(transaction, serverConfigUpdate);
+
   return { config, transaction };
 };
 
+const setGasFeesForTransaction = (
+  transaction: GsnTransactionDetails,
+  serverConfigUpdate: GsnServerConfigPayload
+) => {
+  const serverSuggestedMinPriorityFeePerGas = parseInt(
+    serverConfigUpdate.minMaxPriorityFeePerGas,
+    10
+  );
+
+  const paddedMaxPriority = Math.round(
+    serverSuggestedMinPriorityFeePerGas * 1.4
+  );
+  transaction.maxPriorityFeePerGas = paddedMaxPriority.toString();
+
+  //Special handling for mumbai because of quirk with gas estimate returned by GSN for mumbai
+  if (serverConfigUpdate.chainId === '80001') {
+    transaction.maxFeePerGas = paddedMaxPriority.toString();
```

Comment:
Just making sure I'm understanding this mumbai special case logic correctly (didn't really understand the old deleted comment about it) - it's still basically the same as before, just with padding added, is that right? 

i.e. we're logically just setting maxFeePerGas to minMaxPriorityFeePerGas on mumbai instead of using maxMaxFeePerGas (which would've made more sense, but I guess doesn't work on GSN mumbai?)

Reply:
Yeah hahaha? Seemed like there was some "learned in the real world" lesson here about a special case for mumbai, so I chose to just honor that existing code for now.